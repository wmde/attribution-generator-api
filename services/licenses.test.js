const Licenses = require('./licenses');

const WikiClient = require('./wikiClient');
const parse = require('./parseWikiUrl');

const imageInfo = require('./__fixtures__/imageInfo');
const templates = require('./__fixtures__/templates');

jest.mock('./wikiClient');
jest.mock('./parseWikiUrl');

describe('Licenses', () => {
  const wikiClient = { getResultsFromApi: jest.fn() };

  describe('getLicenseForFile()', () => {
    const url = 'https://en.wikipedia.org/wiki/Apple_Lisa#/media/File:Apple_Lisa.jpg';
    const title = 'File:Apple_Lisa.jpg';
    const wikiUrl = 'https://en.wikipedia.org';
    const fileWikiUrl = 'https://commons.wikipedia.org/';

    beforeEach(() => {
      WikiClient.mockImplementation(() => wikiClient);
      parse.mockImplementationOnce(() => ({ title, wikiUrl }));
      parse.mockImplementationOnce(() => ({ title, wikiUrl: fileWikiUrl }));
    });

    it('gets the file location and returns a matching license based on the page templates', async () => {
      wikiClient.getResultsFromApi
        .mockResolvedValueOnce(imageInfo)
        .mockResolvedValueOnce(templates);

      const service = new Licenses();
      const license = await service.getLicenseForFile(url);

      expect(parse).toHaveBeenCalledWith(url);
      expect(wikiClient.getResultsFromApi).toHaveBeenCalledWith(title, 'imageinfo', wikiUrl, {
        iiprop: 'url',
        iilimit: 1,
        iiurlheight: 300,
      });
      expect(wikiClient.getResultsFromApi).toHaveBeenCalledWith(title, 'templates', fileWikiUrl, {
        tlnamespace: 10,
        tllimit: 100,
      });

      expect(license).toMatchSnapshot();
    });
  });
});
