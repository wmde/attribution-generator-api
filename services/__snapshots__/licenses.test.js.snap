// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Licenses getLicenseForFile() gets the file location and returns a matching license based on the page templates 1`] = `
License {
  "compatibility": Array [
    "cc-by-sa-3.0-de",
    "cc-by-sa-4.0",
  ],
  "groups": Array [
    "cc",
    "cc3",
  ],
  "id": "cc-by-sa-3.0",
  "name": "CC BY-SA 3.0",
  "regexp": /\\^\\(Bild-\\)\\?CC-BY-SA\\(-\\|\\\\/\\)3\\.0\\(\\(\\[\\^-\\]\\+\\.\\+\\|-migrated\\)\\*\\)\\?\\$/i,
  "url": "https://creativecommons.org/licenses/by-sa/3.0/legalcode",
}
`;
